lib/mayaUsd/ufe/UsdUIInfoHandler.cpp

```diff
@@ -71,6 +73,48 @@ bool UsdUIInfoHandler::treeViewCellInfo(const Ufe::SceneItem::Ptr& item, Ufe::Ce
 	return changed;
 }
 
+std::string UsdUIInfoHandler::treeViewIcon(const Ufe::SceneItem::Ptr& item) const
+{
+	// Special case for nullptr input.
+	if (!item) {
+		return "out_USD_UsdTyped.png";	// Default USD icon
+	}
+
+	// We support these node types directly.
+	static const std::map<std::string, std::string> supportedTypes{
```

Comment:
Just to avoid repetition, could we have a really small map with the exceptions (empty string and "AL_MayaReference"), and a set for the rest?  I.e.

static const std::map<std::string, std::string> exceptionsTypes {
{"",					"out_USD_Def.png"},
{"AL_MayaReference",	"out_USD_mayaReference.png"}};
if (inTheExceptionsTypes) {
   return that;
}
static const std::set<std::string> supportedTypes {"BlendShape", "Camera", ...};
if (inSupportedTypes) {
    return "out_USD_" + theType + ".png";
}

Reply:
It is not really repetition as the map key is node type, which are never repeated. I see your point, but I like the current table. It is very clear what icon filename is associated with each node type. What benefit do you see from your suggestion? It actually requires a 2nd search.

Reply:
Don't think the second search is a big deal.  The repetition is in having the type twice, once in the key type, once in the mapped type, when the mapped type is simply the key type with a fixed prefix and a fixed suffix.  Therefore, the set feels more appropriate, because there really is just one value in the associative data structure, the key type.

If you like the way it is now, I don't have that strong an opinion on it, so I'll leave it up to you.